### DISCUSSION TITLE
panda codegen freezes inside Dockerfile

### DISCUSSION THREAD

**Discussion by punkpeye** [chakra-ui/panda#2830] [General]
### Description

When I try to build a docker image that calls `panda codegen`, it seems like it goes in some infinite loop, i.e. never completes.

Outside of `docker build` `panda codegen` completes near immediately.

I am happy to provide more debugging information, but I wasn't sure what would be useful.

I was trying to look for `--verbose` or similar flags, but didn't find anything like that.

I tried `--cpu-prof`, but `panda codegen` never finishes.

### Link to Reproduction

n/a

### Panda CSS Version

0.45.2

### Operating System

- [x] macOS
- [ ] Windows
- [ ] Linux

**Comment by punkpeye** [comment]
After some debugging, I realized that I cannot replicate this if I use `docker build`.

I can only replicate this if I use `fly deploy --local-only`.

Not sure who to tag here. Maybe @wjordan @aschiavo @gwuah (people working on flyctl). I am using 0.2.124.

Their `docker build` abstraction must be using some configuration that breaks `panda codegen`.

I tried adding `--verbose --debug` and similar, but it does not output anything that I could use to advance debugging.

**Comment by punkpeye** [comment]
@segunadebayo any help here? keep running back into this issue

**Reply by punkpeye** [reply]
I ended up just committing the generated files and adding them to Docker image. However, ideally this would be an automated step.

**Reply by punkpeye** [reply]
Locally this same step takes no more than a second, so something is definitely broken when inside Docker.

**Comment by anubra266** [comment]
@punkpeye there was once something like this. I'm no docker expert, but just pointing it out, in case it's related
![CleanShot 2024-09-27 at 21 11 49@2x](https://github.com/user-attachments/assets/1fa6afce-4a4c-4a99-9a9f-7389394b2cb8)
